appvia/kore
# Respect node pool/group field immutable flags on the UI and field visibility changes

## Summary

Adding functionality for GKE, EKS and AKS node pools/groups
Simplifying plan field visibility, for plan and node pools

**Which issue(s) this PR resolves**:

Resolves #1146 

## Checklist

~- [ ] I've created a PR to update the [documentation](https://github.com/appvia/kore-docs): PR LINK~

## Changes

Field with `immutable=true` cannot be edited once the cluster is created
These fields are not shown on the drawer render, along with any fields that are not-editable according to the policy
I've introduced a switch on the node pool drawer, `Show all parameters`, which will display, in disabled mode, the non-editable and immutable field.
If a new node pool/group is created those fields will always be editable, until it's saved
The `Show read-only parameters` checkbox for a cluster plan has been changed to a switch, `Show all parameters` which will simply show the non-editable and immutable (if created) fields.

### Additional changes

* changing some fields on the top level plan render to be disabled when not editable, rather than read-only as this makes them more consistent across the plan
* removing unused `setPrices` function

## Screenshots

**Plan**

<img width="1060" alt="Screen Shot 2020-08-10 at 10 57 35" src="https://user-images.githubusercontent.com/1334068/89771545-633e6700-daf8-11ea-82b0-aeefc1d701db.png">

**Node pool**

<img width="1048" alt="Screen Shot 2020-08-10 at 10 57 53" src="https://user-images.githubusercontent.com/1334068/89771548-65082a80-daf8-11ea-80c1-fb4c4100a599.png">

## Testing

* create/edit cluster plans for all clouds
* edit created cluster node pools, except the immutable fields
* check ability to add new node pools to existing clusters
* show all switch revealing/hiding non-editable and immutable fields on plan and node pools
